test/unit/batcher/TwoWayBatcher.test.ts

```diff
@@ -0,0 +1,408 @@
+import { utils } from 'ethers'
+import { expect } from 'chai'
+import HRE, { waffle } from 'hardhat'
+
+import {
+  IEmptySetReserve__factory,
+  IERC20__factory,
+  TwoWayBatcher,
+  TwoWayBatcher__factory,
+} from '../../../types/generated'
+
+import { MockContract } from '@ethereum-waffle/mock-contract'
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers'
+import { nextContractAddress } from '../../testutil/contract'
+
+const { ethers } = HRE
+
+describe('TwoWayBatcher', () => {
```

Comment:
is it possible to do `close` tests? I know those are pretty hard to setup, but also the scariest code-path

Reply:
There are `close` tests at the bottom of this file. Let me know if there's any cases missed there

Reply:
Oh but we do need integration tests for `close`, I'll add those.